cmd/marker/main.go

```diff
@@ -20,25 +20,39 @@ import (
 	"time"
 
 	"github.com/golang/glog"
+	"github.com/kubevirt/bridge-marker/pkg/cache"
 	"github.com/kubevirt/bridge-marker/pkg/marker"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 func main() {
 	nodeName := flag.String("node-name", "", "name of kubernetes node")
-	const defaultPollInterval = 60
-	pollInterval := flag.Int("update-interval", defaultPollInterval, fmt.Sprintf("interval between updates in seconds, %d by default", defaultPollInterval))
+	const defaultUpdateInterval = 60
+	updateInterval := flag.Int("update-interval", defaultUpdateInterval, fmt.Sprintf("interval between updates in seconds, %d by default", defaultUpdateInterval))
+	const defaultReconcileInterval = 10
+	reconcileInterval := flag.Int("reconcile-interval", defaultReconcileInterval, fmt.Sprintf("interval between node bridges reconcile in minutes, %d by default", defaultReconcileInterval))
 
 	flag.Parse()
 
 	if *nodeName == "" {
 		glog.Fatal("node-name must be set")
 	}
 
-	for {
-		err := marker.Update(*nodeName)
+	cache := cache.Cache{}
+	wait.PollImmediateInfinite(time.Duration(*updateInterval) * time.Second, func() (bool, error) {
+		if time.Now().Sub(cache.LastRefreshTime()) >= time.Duration(*reconcileInterval) * time.Minute {
+			reportedBridges, err := marker.GetReportedResources(*nodeName)
+			if err != nil {
+				glog.Errorf("GetReportedResources failed: %v", err)
```

Comment:
Don't we have to stop here ?

Reply:
I was not sure, maybe it is better to log the error and try again?